src/lib/MakeEvalable.ts

```diff
@@ -0,0 +1,167 @@
+const relIndent = " ".repeat(2);
+
+export interface ConstructorFunc<T extends object> {
+  new (...params: unknown[]): T;
+}
+export type FactoryFunc<T extends object> = (...params: unknown[]) => T;
+export type DestructureFunc<T extends object> = (obj: T) => unknown[];
+
+type CustomTypePlugin<T extends object> = {
+  destructureFunc: DestructureFunc<T>;
+  factory?: FactoryFunc<T>;
+  name?: string;
+};
+const customTypePlugins = new Map<
+  ConstructorFunc<object>,
+  CustomTypePlugin<object>
+>();
+
+class AnnotatedToken {
+  constructor(
+    public text: string,
+    public type: "open" | "close" | "delimiter",
+  ) {}
+}
+type Token = string | AnnotatedToken;
+type TokenGenerator = Generator<Token>;
+
+function* makeAttrListTokens(obj: object): TokenGenerator {
+  for (const [attrName, attrValue] of Object.entries(obj!)) {
+    yield `"${attrName.toString()}": `;
+    yield* makeEvalableTokens(attrValue);
+    yield new AnnotatedToken(",", "delimiter");
+  }
+}
+
+function* makeListTokens(obj: unknown[]): TokenGenerator {
+  for (const item of obj) {
+    yield* makeEvalableTokens(item);
+    yield new AnnotatedToken(",", "delimiter");
+  }
+}
+
+function* makeEvalableTokens(obj: unknown): TokenGenerator {
+  if (obj === null) {
```

Comment:
is `null` not correctly handled by `JSON.stringify` at the end?

Reply:
`typeof null == "object"`

Thus this case would be translated to "{}"